src/App.tsx

```diff
@@ -15,71 +16,90 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+type State = {
+  selectedGood: string,
+};
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+export class App extends React.Component<{}, State> {
+  state = {
+    selectedGood: 'Jam',
+  };
 
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  handleClick = (item: string) => (
+    this.state.selectedGood === item
+      ? this.setState({ selectedGood: '' })
+      : this.setState({ selectedGood: item })
+  );
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+  render() {
+    const { selectedGood } = this.state;
+    const isSelected = (selectedGood.length > 0);
+
+    return (
+      <main className="section container">
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+        <h1 className="title is-flex is-align-items-center">
+          {isSelected
+            ? `${selectedGood} is selected`
+            : 'No goods selected'}
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
+          { (isSelected) && (
+            // eslint-disable-next-line jsx-a11y/control-has-associated-label
             <button
-              data-cy="RemoveButton"
+              data-cy="ClearButton"
               type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+              onClick={() => (
+                this.setState({ selectedGood: '' })
+              )}
+              className="delete ml-3"
+            />
+          )}
+        </h1>
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+        <table className="table">
+          <tbody>
+            {goods.map(item => {
+              const itemcheck = selectedGood === item;
```

Comment:
that's what I meant, your boolean variables should be named as booleans. Here you have `itemcheck`